blocks/init/src/Blocks/components/button/button.php

```diff
@@ -49,46 +58,34 @@
 	Components::selector($buttonIsAnchor, 'js-scroll-to-anchor'),
 ]);
 
+$buttonTag = $buttonUrl ? 'a' : 'button';
 ?>
 
-<?php if (! $buttonUrl) { ?>
-	<button
-		class="<?php echo esc_attr($buttonClass); ?>"
-		<?php if (!empty($buttonId)) { ?>
-			id="<?php echo esc_attr($buttonId); ?>"
-		<?php } ?>
-		title="<?php echo esc_attr($buttonContent); ?>"
-		<?php if (!empty($buttonAriaLabel)) { ?>
-			aria-label="<?php echo esc_attr($buttonAriaLabel); ?>"
-		<?php } ?>
-		data-id="<?php echo esc_attr($unique); ?>"
-		<?php
-		foreach ($buttonAttrs as $key => $value) {
-			echo wp_kses_post("{$key}=" . $value . " ");
-		}
-		?>
-	>
-		<?php echo esc_html($buttonContent); ?>
-	</button>
-
-<?php } else { ?>
-	<a
-		href="<?php echo esc_url($buttonUrl); ?>"
-		class="<?php echo esc_attr($buttonClass); ?>"
-		<?php if (!empty($buttonId)) { ?>
-			id="<?php echo esc_attr($buttonId); ?>"
-		<?php } ?>
-		title="<?php echo esc_attr($buttonContent); ?>"
-		<?php if (!empty($buttonAriaLabel)) { ?>
-			aria-label="<?php echo esc_attr($buttonAriaLabel); ?>"
-		<?php } ?>
-		data-id="<?php echo esc_attr($unique); ?>"
-		<?php
-		foreach ($buttonAttrs as $key => $value) {
+<?php
+	// phpcs:ignore Eightshift.Security.ComponentsEscape.OutputNotEscaped
+	echo Components::outputCssVariables($attributes, $manifest, $unique, $globalManifest);
+?>
+
+<?php // phpcs:ignore Eightshift.Security.ComponentsEscape.OutputNotEscaped ?>
+<<?php echo $buttonTag; ?>
+	class="<?php echo esc_attr($buttonClass); ?>"
+	<?php
+	foreach ($buttonAttrs as $key => $value) {
+		if (!empty($key) && !empty($value)) {
 			echo wp_kses_post("{$key}=" . $value . " ");
 		}
-		?>
-	>
-		<?php echo esc_html($buttonContent); ?>
-	</a>
-<?php } ?>
+	}
+	?>
+>
+	<?php
+	// phpcs:ignore Eightshift.Security.ComponentsEscape.OutputNotEscaped
+	echo Components::render('icon', Components::props('icon', $attributes, [
```

Comment:
Same as the above.

Also the `phpcs.xml.dist` should probably have the public property of that sniff added so that the `overridenClass` property can be set which would identify the `EightshiftBoilerplateVendor\EightshiftLibs\Helpers\Components` as the overridden class.

Reply:
> Also the phpcs.xml.dist should probably have the public property of that sniff added so that the overridenClass property can be set which would identify the EightshiftBoilerplateVendor\EightshiftLibs\Helpers\Components as the overridden class.

🤔 let's leave that for 8.1 😅